pkg/events/check.go

```diff
@@ -110,11 +111,23 @@ func (ce *CheckEvent) GenerateListOfAffectedApps(ctx context.Context, repo *git.
 		matcher = affected_apps.NewMultiMatcher(matcher, configMatcher)
 	}
 
-	ce.affectedItems, err = matcher.AffectedApps(ctx, ce.fileList, targetBranch)
+	// use the changed file path to get the list of affected apps
+	// fileList is a list of changed files in the PR/MR, e.g. ["path/to/file1", "path/to/file2"]
+	ce.affectedItems, err = matcher.AffectedApps(ctx, ce.fileList, targetBranch, repo)
 	if err != nil {
 		telemetry.SetError(span, err, "Get Affected Apps")
 		ce.logger.Error().Err(err).Msg("could not get list of affected apps and appsets")
 	}
+
+	for _, appSet := range ce.affectedItems.ApplicationSets {
+		apps, err := generator.GenerateApplicationSetApps(ctx, appSet, &ce.ctr)
+		if err != nil {
+			ce.logger.Error().Err(err).Msg("could not generate apps from appSet")
+			continue
+		}
+		ce.affectedItems.Applications = append(ce.affectedItems.Applications, apps...)
```

Comment:
On the one hand, we'll end up with duplicates in this slice. On the other hand, the `CheckEvent.queueApp` function (line 314 of the same file) will avoid diffing dupes (around lines 321-325). We should be fine, and I don't think we need to spend the extra effort to avoid the dupes in this slice. Good call out though!

Reply:
(also :+1: on the love for tests!)

Reply:
I had to refactor Check.go 's GenerateListOfAffectedApps quite a bit to make it testable. Added the test just now.